domains/client/domain-operator/src/aux_schema.rs

```diff
@@ -182,8 +218,56 @@ where
                     .as_slice(),
                 best_domain_hash.encode().as_slice(),
             ),
+            (
+                (BEST_DOMAIN_HASH_KEYS, best_domain_number)
+                    .encode()
+                    .as_slice(),
+                domain_hash_keys.encode().as_slice(),
+            ),
         ],
         vec![],
+    )?;
+
+    cleanup_domain_hash_and_consensus_hash::<_, Block, CBlock>(domain_client)
+}
+
+fn cleanup_domain_hash_and_consensus_hash<Client, Block, CBlock>(
+    domain_client: &Arc<Client>,
+) -> ClientResult<()>
+where
+    CBlock: BlockT,
+    Block: BlockT,
+    Client: HeaderBackend<Block> + AuxStore,
+{
+    let mut finalized_domain_number = domain_client.info().finalized_number;
+
+    let mut deletions = vec![];
+    while finalized_domain_number > Zero::zero() {
```

Comment:
It seems like this loop could get more expensive as the latest finalized block gets higher. How expensive is loading and decoding an empty entry from the backend?

Reply:
this is done everytime domain imports new blocks. So for domains it will be always one.

For taurus, for the first run since we dont finalize yet, will be a big number and next call after will be one. Having said that, since this is a new aux for taurus, there wont be any storage items in it.